### PR TITLE
Fix delete on clause may not work in routineLoad

### LINKED ISSUE #4682: [BUG] delete on in batch delete  may not work
**Describe the bug**
* delete on in routine load not work when not using column maping
* append and delete with delete on  error message is different in stream load and routine load

### PR DESCRIPTION
**Pull request by yangzhg**
## Proposed changes

fix delete on  may not work in some cases, this is describe in #4682 

## Types of changes

What types of changes does your code introduce to Doris?
_Put an `x` in the boxes that apply_

- [x] Bugfix (non-breaking change which fixes an issue)
- [] New feature (non-breaking change which adds functionality)
- [] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [] Documentation Update (if none of the other choices apply)
- [] Code refactor (Modify the code structure, format the code, etc...)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I have create an issue on (Fix  #4682), and have described the bug/feature there in detail
- [x] Compiling and unit tests pass locally with my changes
- [] I have added tests that prove my fix is effective or that my feature works
- [] If this change need a document change, I have updated the document
- [x] Any dependent changes have been merged

## Further comments

If this is a relatively large or complex change, kick off the discussion at dev@doris.apache.org by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: fe/fe-core/src/main/java/org/apache/doris/analysis/CreateRoutineLoadStmt.java [hunk 1]
```diff
@@ -297,6 +311,19 @@ public void checkDBTable(Analyzer analyzer) throws AnalysisException {
         if (Strings.isNullOrEmpty(tableName)) {
             throw new AnalysisException("Table name should not be null");
         }
+        Database db = Catalog.getCurrentCatalog().getDb(dbName);
+        if (db == null) {
+            throw new AnalysisException("database: " + dbName + "not  found.");
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/analysis/CreateRoutineLoadStmt.java:316]:
```suggestion
            throw new AnalysisException("database: " + dbName + " not found.");
```

File: fe/fe-core/src/main/java/org/apache/doris/load/BrokerFileGroup.java [hunk 2]
```diff
@@ -385,6 +385,13 @@ public void write(DataOutput out) throws IOException {
         // src table
         out.writeLong(srcTableId);
         out.writeBoolean(isLoadFromTable);
+        Text.writeString(out, mergeType.name());
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/load/BrokerFileGroup.java:388]: I think we can change this class to GSON DeSer.

Comment (yangzhg) [reply fe/fe-core/src/main/java/org/apache/doris/load/BrokerFileGroup.java:388]: this method is useless， I have marked it  as  depricated

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**